### PR TITLE
BUG: avoid specifying default coerce_timestamps in to_parquet

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Looking into the usage question of https://github.com/pandas-dev/pandas/issues/31572, I noticed that specifying the version to allow writing nanoseconds to parquet worked in plain pyarrow code, but not with pandas' `to_parquet`. 
This is because we hardcode `coerce_timestamps="ms"` while the default is `None`, which has version-dependent behaviour (eg if version="2.0", actually write the nanosecond data)

### CODE REVIEW

File: pandas/tests/io/test_parquet.py [hunk 1]
```diff
@@ -564,6 +564,13 @@ def test_additional_extension_types(self, pa):
         )
         check_round_trip(df, pa)
 
+    @td.skip_if_no("pyarrow", min_version="0.14")
+    def test_timestamp_nanoseconds(self, pa):
+        # with version 2.0, pyarrow defaults to writing the nanoseconds, so
+        # this should work with error
```
Comment (TomAugspurger) [inline pandas/tests/io/test_parquet.py]: Should this say "work **without** error"?

Comment (jorisvandenbossche) [reply pandas/tests/io/test_parquet.py]: yes, of course ... !

File: doc/source/whatsnew/v1.1.0.rst [hunk 2]
```diff
@@ -180,7 +180,9 @@ I/O
 - Bug in :meth:`read_json` where integer overflow was occuring when json contains big number strings. (:issue:`30320`)
 - `read_csv` will now raise a ``ValueError`` when the arguments `header` and `prefix` both are not `None`. (:issue:`27394`)
 - Bug in :meth:`DataFrame.to_json` was raising ``NotFoundError`` when ``path_or_buf`` was an S3 URI (:issue:`28375`)
--
+- Bug in :meth:`DataFrame.to_parquet` overwriting pyarrow's default for
+  ``coerce_timestamps``; following pyarrow's default allow to write nanosecond
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v1.1.0.rst]:
```suggestion
  ``coerce_timestamps``; following pyarrow's default allows writing nanosecond
```

**Review by TomAugspurger — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jorisvandenbossche

**Comment by mnylen** [comment]
For us this change caused a nasty bug after upgrading from pandas 1.0.3 to 1.1.x. Apparently AWS Athena/Presto doesn't support nanosecond precision, so our timestamps started appearing with year 52000 when we created the parquet files using pandas 1.1.x. Just FYI for anyone else having the same issue.

**Comment by findepi** [comment]
@mnylen FWIW, since recently Presto does support nanosecond precision for date/time types, including timestamps. (see https://github.com/prestosql/presto/issues/1284 for more info).
However, this seems to be a different problem. If you can time, would you be able to answer a question in https://github.com/prestosql/presto/issues/4662#issuecomment-667583564 so that we know what needs to be improved?

**Comment by mnylen** [comment]
@findepi I mentioned Presto only because AWS Athena uses that underneath. Athena is based on Presto 0.172, so it's likely that even if Presto nowadays supports nanoseconds for timestamps, it'll take a long time before Athena upgrades to the newer version.

**Pull request merged**